internal/test/e2e/run.go

```diff
@@ -65,7 +66,11 @@ func RunTestsInParallel(conf ParallelRunConf) error {
 
 	instancesConf := splitTests(testsList, conf)
 	logTestGroups(instancesConf)
+	maxConcurrentTests := conf.MaxConcurrentTests
+	// Add a blocking channel to only allow for certain number of tests to run at a time
+	queue := make(chan struct{}, maxConcurrentTests)
 	for _, instanceConf := range instancesConf {
+		queue <- struct{}{}
```

Comment:
We should probably look into a fan out pattern (feeding tests to a pool of routines through a channel) as opposed to use a channel to channel just to block the main loop